lib/galaxy/tools/parameters/validation.py

```diff
@@ -248,6 +249,23 @@ def validate( self, value, trans=None ):
                 raise ValueError( self.message )
 
 
+class DatasetFilesPathEmptyValidator( Validator ):
+    """Validator that checks if a dataset's files_path exists and is not empty."""
+    def __init__( self, message=None ):
+        self.message = message
+
+    @classmethod
+    def from_element( cls, param, elem ):
+        return cls( elem.get( 'message', None ) )
+
+    def validate( self, value, trans=None ):
+        if value:
+            if not os.path.isdir(value.extra_files_path) or len(os.listdir(value.extra_files_path)) == 0:
```

Comment:
Thanks for the quick response on this.  Just to confirm, since these abstractions generally relate to files instead of directories, I still need to check if the path is a valid directory, right?
```
>>> p1 = '/not/a/path'
>>> os.path.getsize(p1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/scratch/biotools/galaxy/galaxy/.venv/lib/python2.7/genericpath.py", line 49, in getsize
    return os.stat(filename).st_size
OSError: [Errno 2] No such file or directory: '/not/a/path'
```
So the only change would be to this line: https://github.com/galaxyproject/galaxy/pull/3918/files#diff-cbfb6ffd518f2a68bf40004cfe889947R263, which could become:
```
if not os.path.isdir(value.extra_files_path) or value.get_size() == 0:
```
Is this correct?